src/main/java/fr/insee/genesis/controller/adapter/LunaticXmlAdapter.java

```diff
@@ -39,20 +86,63 @@ public static SurveyUnitUpdateDto convert(LunaticXmlSurveyUnit su, VariablesMap
                             .build());
                 }
             }
-        });
+        }
         surveyUnitUpdateDto.setVariablesUpdate(variablesUpdate);
 
+        //External variables goes into the COLLECTED DTO
         List<VariableDto> externalVariables = new ArrayList<>();
         su.getData().getExternal().forEach(lunaticXmlExternalData ->
-            externalVariables.add(VariableDto.builder()
-                    .idVar(lunaticXmlExternalData.getVariableName())
-                    .values(transformToList(lunaticXmlExternalData.getValues().get(0).getValue()))
-                    .build())
+                externalVariables.add(VariableDto.builder()
+                        .idVar(lunaticXmlExternalData.getVariableName())
+                        .values(transformToList(lunaticXmlExternalData.getValues().get(0).getValue()))
+                        .build())
         );
         surveyUnitUpdateDto.setExternalVariables(externalVariables);
+
         return surveyUnitUpdateDto;
     }
 
+    private static SurveyUnitUpdateDto getOtherDataFromSurveyUnit(LunaticXmlSurveyUnit su, SurveyUnitUpdateDto surveyUnitUpdateDto, VariablesMap variablesMap, DataState dataState) {
```

Comment:
This method should treat all 5 dataState (with also DataState.COLLECTED), and should be named getCollectedData.